boards/mamba/f4mk2/init/rc.board_extras

```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+#
+# OmnibusF4SD specific board extras init
+#------------------------------------------------------------------------------
+
+if ! param compare OSD_ATXXXX_CFG 0
+then
+	atxxxx start -s
+fi
+
+if ! param compare MAV_2_CONFIG 0
+then
+	serdis
+	sleep 1;ekf2 stop;sleep 1;ekf2 start
```

Comment:
Why?

Reply:
... have not yet found a better solution to have enough ram left when GPS and Wifi telemetry is active.
With too little ram it flies despite that, but no home position gets set.  I would like to allow USB only shortly after the boot. And restarting ekf2 lowers acc0 fail / acc0 high bias errors and sets alt rel to zero.

Reply:
Is there a way to use the result of the amount of free ram with "FREE" in a query?
Then serdis could only be triggered if a critical treshold of free ram is reached.

Reply:
Take a look at https://github.com/PX4/PX4-Autopilot/pull/16180 where mavlink USB is launched automatically if VBUS is connected.

From there we could add in the memory check or other options.

Reply:
tried #16180 , works great with the new mamba target. (will try it with other targets e.g. PixRacer)
USB disconnected frees up 14 k more than using only serdis.
USB connected now needs about 8 k more than before. 
Below a certain free ram value, no more parameters can be saved, which was actually my original main reason. This would saw off the branch you are sitting on if you screw around with telemetry 2 with my variant.
For all this #16180 is the perfect solution.
What was the reason why this was not merged earlier?

Reply:
There is only one point where it does not work completely: For the complete recognition of the GPS signal, it is necessary to supply the copter with power via an usb power bank, otherwise the lipo would be drained before the flight from the video unit. In this case, USB should also be recognised as completely disconnected but it is not.

I have also found this to be the case when testing the PixRacer, where I usually carry out Wifi telemetry tests with a powerbank connected to the USB.

Reply:
I still have a question about the w25q128.
Is the mtd driver not included due to "constrained flash", can't do anything with the 16Mb size, or can it only do qspi since there is only a standard spi connection to this flash on the board?

Reply:
> Is the mtd driver not included due to "constrained flash", can't do anything with the 16Mb size, or can it only do qspi since there is only a standard spi connection to this flash on the board?

I'm not quite sure what you're asking, but so far `mtd` has been mainly used for our ramtron or at24xxx (i2c eeprom) use case with parameters. https://github.com/PX4/PX4-Autopilot/blob/c9b5e488f50561be0aeb9a50d59d09514e3dba0a/boards/px4/fmu-v5x/src/mtd.cpp#L37-L102

Reply:
I know most boards use a 32 k spi fram. But I thought there might be a way to use that 16Mb flash with the W25 or the W25QXXXJV nuttX driver to handle a few waypoints since all 20x20mm boards don't offer a sd slot.

Reply:
It's very easy. I have been successfully using MX25L 4MB for quite a time.
In fact just add attach function in px4_mtd. Take a look for my patch as reference. It is not for recent master, but you should get a point.
I think there will no problem to do the same with W25.

Reply:
@lukegluke yes it's definitely worth it if people are using it.